README.md

```
@@ -180,16 +286,160 @@ flowchart TD
+| `POST` | `/api/move/` | Execute a move on the board | `/api/move/` |
+| `GET` | `/api/valid-moves/` | Get all legal moves for a selected piece | `/api/valid-moves/?row=6&col=4` |
+| `POST` | `/api/new-game/` | Start a new PvP or PvE game | `/api/new-game/` |
+| `GET` | `/api/check-promotion/` | Check if a pawn reaches the promotion rank | `/api/check-promotion/?from_row=1&from_col=0&to_row=0` |
+| `POST` | `/api/ai-move/` | Request the engine to compute the best move | `/api/ai-move/` |
+| `POST` | `/api/pause/` | Pause/Resume game timer countdown | `/api/pause/` |
+
+---
+
+### Request/Response JSON Examples
+
+#### 1. Retrieve Game State (`GET /api/state/`)
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)